[exec] Unable to use %2$s

## Related to
This bug is related to a discussion in the forums:
https://community.openhab.org/t/an-exception-occurred-while-formatting-the-command-line/33385

## Expected
It is expected that the value of the String item is replaces the %2$s on the command line of and exec2 Thing.

## Issue
As soon as the %2$s is added, the command no longer executes.
I experience this error:
> 2017-08-15 20:07:12.902 [ERROR] [hab.binding.exec.handler.ExecHandler] - An exception occurred while formatting the command line with the current time and input values : ‘Format specifier ‘%2$s’’

Using %1$s works as it adds the date/time as expected.

I have attempted to create things & items using habmin, and actual files with no success.
The value of the input string has been set.

## Steps and settings
The simplest setting i have tried are:

> items/demo.items:String Output4 “Output4 [%s]” (Command) { channel=“exec:command:echo_exec4:output” }
> items/demo.items:String Input4 “Input4 [%s]” (Command) { channel=“exec:command:echo_exec4:input” }
> things/demo.things:Thing exec:command:echo_exec4 [command="/bin/echo abc ‘%2$s’", interval=30, autorun=true]

## Context
I am trying to replicate behavior of exec1 where there was a separate command for ON, OFF and GET. By using the %2$s, I should be able to pass the state of my switch to the command line.

## Your Environment
I am running OpenHAB 2.1.0
Running on Ubuntu xenial.
openjdk version "1.8.0_131"
OpenJDK Runtime Environment (build 1.8.0_131-8u131-b11-2ubuntu1.16.04.3-b11)
OpenJDK 64-Bit Server VM (build 25.131-b11, mixed mode)

--- a/bundles/org.openhab.binding.exec/README.md
+++ b/bundles/org.openhab.binding.exec/README.md
@@ -67,7 +67,7 @@ All Things support the following channels:
 | run             | Switch    | Send ON to execute the command, the current state tells whether it is running or not |
 | lastexecution   | DateTime  | Time/Date the command was last executed, in yyyy-MM-dd'T'HH:mm:ss.SSSZ format        |
 
-**Attention:** Linking `input` to any other item type than `String` will result in errorneous behaviour.
+**Attention:** Linking `input` to any other item type than `String` will result in erroneous behavior.
 If needed, please use a rule to convert your item's state to a string.
 Also note that only commands (e.g. `sendCommand`) to the `input` channel are recognized, updating the item's state will not work (e.g. `postUpdate`).
 

--- a/bundles/org.openhab.binding.exec/README.md
+++ b/bundles/org.openhab.binding.exec/README.md
@@ -68,7 +68,7 @@ All Things support the following channels:
 | lastexecution   | DateTime  | Time/Date the command was last executed, in yyyy-MM-dd'T'HH:mm:ss.SSSZ format        |
 
 **Attention:** Linking `input` to any other item type than `String` will result in errorneous behaviour.
-If needed, please use rule to convert your item state to a string.
+If needed, please use a rule to convert your item's state to a string.
 Also note that only commands (e.g. `sendCommand`) to the `input` channel are recognized, updating the item's state will not work (e.g. `postUpdate`).
 
 ## Minimal Example

--- a/bundles/org.openhab.binding.exec/README.md
+++ b/bundles/org.openhab.binding.exec/README.md
@@ -67,6 +67,9 @@ All Things support the following channels:
 | run             | Switch    | Send ON to execute the command, the current state tells whether it is running or not |
 | lastexecution   | DateTime  | Time/Date the command was last executed, in yyyy-MM-dd'T'HH:mm:ss.SSSZ format        |
 
+**Attention:** Linking `input` to any other item type than `String` will result in errorneous behaviour.
+If needed, please use rule to convert your item state to a string.
+Also note that only commands (e.g. `sendCommand`) to the `input` channel are recognized, updating the item's state will not work (e.g. `postUpdate`).
 
 ## Minimal Example
 

--- a/bundles/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/internal/handler/ExecHandler.java
+++ b/bundles/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/internal/handler/ExecHandler.java
@@ -21,6 +21,7 @@ import java.math.BigDecimal;
 import java.time.ZonedDateTime;
 import java.util.Arrays;
 import java.util.Calendar;
+import java.util.Date;
 import java.util.IllegalFormatException;
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
@@ -168,16 +169,17 @@ public class ExecHandler extends BaseThingHandler {
             // problem for external commands that generate a lot of output, but this will be dependent on the limits
             // of the underlying operating system.
 
+            Date date = Calendar.getInstance().getTime();
             try {
                 if (lastInput != null) {
-                    commandLine = String.format(commandLine, Calendar.getInstance().getTime(), lastInput);
+                    commandLine = String.format(commandLine, date, lastInput);
                 } else {
-                    commandLine = String.format(commandLine, Calendar.getInstance().getTime());
+                    commandLine = String.format(commandLine, date);
                 }
             } catch (IllegalFormatException e) {
                 logger.warn(
-                        "An exception occurred while formatting the command line with the current time and input values : '{}'",
-                        e.getMessage());
+                        "An exception occurred while formatting the command line '{}' with the current time '{}' and input value '{}': {}",
+                        commandLine, date, lastInput, e.getMessage());
                 updateState(RUN, OnOffType.OFF);
                 updateState(OUTPUT, new StringType(e.getMessage()));
                 return;

